pkg/cmd/project/unlink/unlink.go

```diff
@@ -0,0 +1,195 @@
+package unlink
+
+import (
+	"fmt"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmd/project/shared/client"
+	"github.com/cli/cli/v2/pkg/cmd/project/shared/queries"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/shurcooL/githubv4"
+	"github.com/spf13/cobra"
+	"net/http"
+	"strconv"
+)
+
+type unlinkOpts struct {
+	number    int32
+	owner     string
+	repo      string
+	team      string
+	projectID string
+	repoID    string
+	teamID    string
+	format    string
+	exporter  cmdutil.Exporter
+}
+
+type unlinkConfig struct {
+	httpClient func() (*http.Client, error)
+	client     *queries.Client
+	opts       unlinkOpts
+	io         *iostreams.IOStreams
+}
+
+type unlinkProjectFromRepoMutation struct {
+	UnlinkProjectV2FromRepository struct {
+		Repository queries.Repository `graphql:"repository"`
+	} `graphql:"unlinkProjectV2FromRepository(input:$input)"`
+}
+
+type unlinkProjectFromTeamMutation struct {
+	UnlinkProjectV2FromTeam struct {
+		Team queries.Team `graphql:"team"`
+	} `graphql:"unlinkProjectV2FromTeam(input:$input)"`
+}
+
+func NewCmdUnlink(f *cmdutil.Factory, runF func(config unlinkConfig) error) *cobra.Command {
+	opts := unlinkOpts{}
+	linkCmd := &cobra.Command{
+		Short: "Unlink a project from a repository or a team",
+		Use:   "unlink [<number>] [flag]",
+		Example: heredoc.Doc(`
+			# unlink monalisa's project 1 from her repository "my_repo"
+			gh project unlink 1 --owner monalisa --repo my_repo
+
+			# unlink monalisa's organization's project 1 from her team "my_team"
+			gh project unlink 1 --owner my_organization --team my_team
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := client.New(f)
+			if err != nil {
+				return err
+			}
+
+			if len(args) == 1 {
+				num, err := strconv.ParseInt(args[0], 10, 32)
+				if err != nil {
+					return cmdutil.FlagErrorf("invalid number: %v", args[0])
+				}
+				opts.number = int32(num)
+			}
+
+			config := unlinkConfig{
+				httpClient: f.HttpClient,
+				client:     client,
+				opts:       opts,
+				io:         f.IOStreams,
+			}
+
+			if config.opts.repo != "" && config.opts.team != "" {
+				return fmt.Errorf("specify only one repo or team")
+			} else if config.opts.repo == "" && config.opts.team == "" {
+				return fmt.Errorf("specify at least one repo or team")
+			}
```

Comment:
Bringing this inline with the suggestion above:

```suggestion
			if config.opts.repo != "" && config.opts.team != "" {
				return fmt.Errorf("specify only one of `--repo` or `--team`")
			} else if config.opts.repo == "" && config.opts.team == "" {
				return fmt.Errorf("specify either `--repo` or `--team`")
			}
```